internal/server/data/passwordresettoken.go

```diff
@@ -30,6 +30,7 @@ retry:
 
 	tries++
 	if err = save(db, prt); err != nil {
+		// TODO: must use errors.As for error types
 		if tries <= 3 && errors.Is(err, UniqueConstraintError{}) {
```

Comment:
I believe `errors.Is` only works on typed errors if they implement `Is`. We could add an `Is` method to `UniqueConstraintError`, or we should use `errors.As` here.